rmf_site_editor/src/site/drawing.rs

```diff
@@ -0,0 +1,137 @@
+    asset_server: Res<AssetServer>,
+    mut loading_drawings: ResMut<LoadingDrawings>,
+    current_site: Res<CurrentSite>,
+    site_files: Query<(Entity, &DefaultFile)>,
+) {
+    let file_path = get_current_site_path(current_site, site_files);
+    if file_path.is_none() {
+        return;
+    }
+    for (e, source, pose, pixels_per_meter) in &new_drawings {
+        let texture_path = match source {
+            AssetSource::Local(name) => file_path.as_ref().unwrap().with_file_name(name),
+        };
+        let texture_handle: Handle<Image> = asset_server.load(texture_path);
+        (*loading_drawings)
+            .0
+            .insert(texture_handle, (e, pose.clone(), pixels_per_meter.clone()));
+    }
+}
+
+// Asset event handler for loaded drawings
+pub fn handle_loaded_drawing(
+    mut commands: Commands,
+    mut ev_asset: EventReader<AssetEvent<Image>>,
+    assets: Res<Assets<Image>>,
+    mut loading_drawings: ResMut<LoadingDrawings>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    for ev in ev_asset.iter() {
+        if let AssetEvent::Created { handle } = ev {
+            if let Some((entity, pose, pixels_per_meter)) = (*loading_drawings).0.remove(handle) {
+                let img = assets.get(handle).unwrap();
+                let width = img.texture_descriptor.size.width as f32;
+                let height = img.texture_descriptor.size.height as f32;
+                let mut mesh = Mesh::from(make_flat_rectangle_mesh(
+                    height / pixels_per_meter.0,
+                    width / pixels_per_meter.0,
+                ));
+                let uvs: Vec<[f32; 2]> = [[1.0, 1.0], [1.0, 0.0], [0.0, 0.0], [0.0, 1.0]]
+                    .into_iter()
+                    .cycle()
+                    .take(8)
+                    .collect();
+                // TODO Actual Z layering instead of hardcoded Z
+                let mut pose = pose.clone();
+                pose.trans[2] -= 0.01;
+                mesh.insert_attribute(Mesh::ATTRIBUTE_UV_0, uvs);
+                commands
+                    .entity(entity.clone())
+                    .insert_bundle(PbrBundle {
+                        mesh: meshes.add(mesh),
+                        material: materials.add(StandardMaterial {
+                            base_color_texture: Some(handle.clone()),
+                            ..default()
+                        }),
+                        transform: pose.transform(),
+                        ..default()
+                    })
+                    .insert(Selectable::new(entity))
+                    .insert(Category("Drawing".to_string()));
+            }
+        }
+    }
+}
+
+pub fn update_drawing_visuals(
+    changed_drawings: Query<
+        (Entity, &AssetSource, &Pose, &PixelsPerMeter),
+        Or<(Changed<AssetSource>, Changed<PixelsPerMeter>)>,
```

Comment:
Done! https://github.com/open-rmf/rmf_site/pull/73/commits/f030ed24522f72ff17d6232f163c409412734435 I had to tweak the behavior of the system that updates `Changed<Pose>` components since it would silently drop transform scale. As a result of this change pixels per meter change is _a lot_ smoother and faster, thanks for the feedback!

Reply:
The change you made for `Changed<Pose>` is definitely what should've been there in the first place :+1: